src/freenet/pluginmanager/FredPluginFCPMessageHandler.java

```diff
@@ -0,0 +1,580 @@
+         * 
+         * You <b>must not</b> use this for constructing outgoing messages in server or client
+         * implementations.<br>
+         * 
+         * This function is typically to construct incoming messages for passing them to the message
+         * handling function
+         * {@link FredPluginFCPMessageHandler#handlePluginFCPMessage(FCPPluginClient,
+         * FCPPluginMessage)}.<br><br>
+         * 
+         * See the JavaDoc of the member variables with the same name as the parameters for an
+         * explanation of the parameters.<br>
+         */
+        public static FCPPluginMessage constructRawMessage(ClientPermissions permissions,
+                String identifier, SimpleFieldSet params, Bucket data, Boolean success,
+                String errorCode, String errorMessage) {
+            
+            return new FCPPluginMessage(permissions, identifier, params, data, success,
+                errorCode, errorMessage);
+        }
+
+        @Override
+        public String toString() {
+            return super.toString() +
+                " (permissions: " + permissions +
+                "; identifier: " + identifier +
+                "; data: " + data +
+                "; success: " + success +
+                "; errorCode: " + errorCode +
+                "; errorMessage: " + errorMessage + 
+                // At the end because a SimpleFieldSet usually contains multiple line breaks.
+                "; params: " + '\n' + (params != null ? params.toOrderedString() : null);
+        }
+    }
+
+
+    /**
+     * Message handling function for messages received from a plugin FCP server or client.<br/><br/>
+     * 
+     * <b>ATTENTION</b>: Please read the different constraints for server and client side message
+     * handlers at the child interfaces:<br/>
+     * - {@link ServerSideFCPMessageHandler#handlePluginFCPMessage(FCPPluginClient,
+     *   FCPPluginMessage)}<br/>
+     * - {@link ClientSideFCPMessageHandler#handlePluginFCPMessage(FCPPluginClient,
+     *   FCPPluginMessage)}<br/>
+     * 
+     * To stress those different constraints, you should also not implement this interface but one
+     * of the child interfaces {@link ServerSideFCPMessageHandler} and
+     * {@link ClientSideFCPMessageHandler}.
+     */
+    FCPPluginMessage handlePluginFCPMessage(FCPPluginClient client, FCPPluginMessage message);
+
+
+    /**
+     * Plugins which provide FCP services to clients must implement this interface.<br/>
+     * The purpose of this interface is to provide a message handling function for processing
+     * messages received from the clients.
+     * 
+     * @see FredPluginFCPMessageHandler
+     *          The parent interface FredPluginFCPMessageHandler provides an overview.
+     * @see ClientSideFCPMessageHandler
+     *          The opposite version of this interface for client plugins
+     */
+    public interface ServerSideFCPMessageHandler extends FredPluginFCPMessageHandler {
```

Comment:
ServerSideFCPMessageHandler, and its counterpart ClientSideFCPMessageHandler, are nested in this class for the primary purpose of allowing the developer to have an overview of the "flow" of messages, and the different restrictions for servers/clients, in one place.
I would like to keep it that way.
I would suggest that I alleviate the issue of the size of this file by instead moving the member class FCPPluginMessage to its own file.
ACK?